### PR TITLE
Added  tests for the keycloak service proxy

### PR DESCRIPTION
**Pull request by ziccardi**
## Description
[keycloak_kafka_test.go](https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/internal/kafka/internal/services/keycloak_kafka_test.go) was duplicating most of the tests already done in [keycloak_service_test](https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/pkg/services/sso/keycloak_service_test.go).
This PR deletes the [keycloak_kafka_test.go](https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/main/internal/kafka/internal/services/keycloak_kafka_test.go) and introduces the new `keycloak_service_proxy_test.go` that tests the service proxy.

With this change, the coverage doesn't change much, but we move the tests in the right package (before we had the test in the `service` package testing objects in the `sso` package) and we can remove methods we created just to expose internals to the tests.

## Checklist (Definition of Done)

- [ ] All acceptance criteria specified in JIRA have been completed
- [ ] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)
- [ ] Documentation added for the feature
- [ ] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer
- [ ] All PR comments are resolved either by addressing them or creating follow up tasks
- [ ] Required metrics/dashboards/alerts have been added (or PR created).
- [ ] Required Standard Operating Procedure (SOP) is added.
- [ ] JIRA has been created for changes required on the client side

### CODE REVIEW

File: pkg/services/sso/keycloak_service_builder.go [hunk 1]
```diff
@@ -101,12 +101,20 @@ func build(providerName string, keycloakConfig *keycloak.KeycloakConfig, realmCo
 	if providerName == keycloak.MAS_SSO ||
 		realmConfig != nil {
 		_, realmConfig := arrays.FindFirst(notNilPredicate, realmConfig, keycloakConfig.KafkaRealm)
-		return newKeycloakService(keycloakConfig, realmConfig.(*keycloak.KeycloakRealmConfig))
```
Comment (machi1990) [inline pkg/services/sso/keycloak_service_builder.go:104]: is this func used anywhere? If not it can be deleted.

File: pkg/services/sso/keycloak_service_proxy.go [hunk 2]
```diff
@@ -10,13 +10,11 @@ import (
 	"github.com/openshift-online/ocm-sdk-go/authentication"
 )
 
-type tokenProvider interface {
-	GetToken() (string, error)
-}
+type tokenProvider func() (string, error)
```
Comment (machi1990) [inline pkg/services/sso/keycloak_service_proxy.go:13]: Why this change?

File: pkg/services/sso/keycloak_service_builder.go (hunk 1 above)
Comment (ziccardi) [reply pkg/services/sso/keycloak_service_builder.go:104]: Good point. Was used by one test, but I replaced it with the builder. Removed.

File: pkg/services/sso/keycloak_service_proxy.go (hunk 2 above)
Comment (ziccardi) [reply pkg/services/sso/keycloak_service_proxy.go:13]: Because the `tokenProvider` is just one method interface: I find creating an interface to pass an object with just one method to be much _javish_ and a bit overkill provided that in go we can pass functions.

Comment (machi1990) [reply pkg/services/sso/keycloak_service_proxy.go:13]: Thanks for the explanation.
